Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions-details.tsx

```diff
@@ -2,53 +2,48 @@ import React, { createContext, useCallback, useContext, useEffect, useMemo, useS
 import isNil from 'lodash/isNil';
 
 import { DEFAULT_PROBLEM_RESULTS_TAKE_CONTESTS_PAGE } from '../../common/constants';
-import { IValidationType } from '../../common/types';
+import { IException } from '../../common/types';
 import {
     IGetSubmissionDetailsByIdUrlParams,
-    IGetSubmissionResultsByProblemAndUserUrlParams,
     IGetSubmissionResultsByProblemUrlParams,
 } from '../../common/url-types';
 import { IHaveChildrenProps } from '../../components/common/Props';
-import { useAuth } from '../use-auth';
 import { useHttp } from '../use-http';
 import { useLoading } from '../use-loading';
 import { useUrls } from '../use-urls';
 
-import { ISubmissionDetails, ISubmissionDetailsType, ISubmissionType, ITestRunType } from './types';
+import {
+    ISubmissionDetails,
+    ISubmissionDetailsType,
+    ISubmissionType,
+    ITestRunType,
+} from './types';
 
 interface ISubmissionsDetailsContext {
     state: {
         currentSubmission: ISubmissionDetailsType | null;
         currentProblemSubmissionResults: ISubmissionDetails[];
-        validationResult: IValidationType;
+        problemResultsError: IException | null;
+        submissionDetailsError: IException | null;
     };
     actions: {
         selectSubmissionById: (submissionId: number) => void;
         getDetails: (submissionId: number) => Promise<void>;
-        getSubmissionResults: (problemId: number, isOfficial: boolean, userId: string) => Promise<void>;
+        getSubmissionResults: (problemId: number, isOfficial: boolean) => Promise<void>;
     };
 }
 
-const defaultState = {
-    state: {
-        currentProblemSubmissionResults: [] as ISubmissionDetails[],
-        validationResult: {
-            message: '',
-            isValid: true,
-            propertyName: '',
-        },
-    },
-};
+const defaultState = { state: { currentProblemSubmissionResults: [] as ISubmissionDetails[] } };
 
 const SubmissionsDetailsContext = createContext<ISubmissionsDetailsContext>(defaultState as ISubmissionsDetailsContext);
 
 type ISubmissionsDetailsProviderProps = IHaveChildrenProps
 
 const SubmissionsDetailsProvider = ({ children }: ISubmissionsDetailsProviderProps) => {
-    const { state: { user } } = useAuth();
     const { startLoading, stopLoading } = useLoading();
     const [ currentSubmissionId, selectSubmissionById ] = useState<number>();
-    const [ validationResult, setValidationResult ] = useState<IValidationType>(defaultState.state.validationResult);
+    const [ problemResultsError, setProblemResultsError ] = useState<IException | null>(null);
+    const [ submissionDetailsError, setSubmissionDetailsError ] = useState<IException | null>(null);
```

Comment:
The rendering logic of these is the same. It can be made into an array and then you can use an array to avoid double checks

Reply:
Done.